### ISSUE TITLE
Do not use simple constant property patterns as conditions of 'if' statements

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#939]
We should issue a warning in case someone writes code like
```c#
if (xyz is { Enabled: true })
{
    ...
}
```

and propose to change it into code like
```c#
if (xyz.Enabled is true)
{
    ...
}
```
or
```c#
if (xyz?.Enabled is true)
{
    ...
}
```
The reason is that the latter ones are easier to understand while the first one over-uses pattern matching.

**Note:** This is _no issue_ inside _switch expression arms_.
